vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -66,23 +66,46 @@ type ClusterStateFeeder interface {
 	GarbageCollectCheckpoints()
 }
 
-// NewClusterStateFeeder creates new ClusterStateFeeder with internal data providers, based on kube client config and a historyProvider.
-func NewClusterStateFeeder(config *rest.Config, clusterState *model.ClusterState) ClusterStateFeeder {
-	kubeClient := kube_client.NewForConfigOrDie(config)
-	oomObserver := oom.NewObserver()
-	podLister := newPodClients(kubeClient, &oomObserver)
-	watchEvictionEventsWithRetries(kubeClient, &oomObserver)
+// ClusterStateFeederMaker makes instances of ClusterStateFeeder.
+type ClusterStateFeederMaker struct {
```

Comment:
Rename Makers to Factories. Here and elsewhere.

Reply:
Done.